python/podio/frame.py

```diff
@@ -107,9 +122,27 @@ def get(self, name):
       raise KeyError(f"Collection '{name}' is not available")
     return collection
 
+  def put(self, collection, name):
+    """Put the collection into the frame
+
+    Args:
+        collection (podio.CollectionBase): The collection to put into the Frame
+        name (str): The name of the collection
+
+    Returns:
+        podio.CollectionBase: The reference to the collection that has been put
+           into the Frame
+
+    Raises:
+        ValueError: If collection is not actually a podio.CollectionBase
+    """
+    if not _is_collection_base(collection):
+      raise ValueError("Can only put podio collections into a Frame")
+    return self._frame.put(cppyy.gbl.std.move(collection), name)
```

Comment:
I just tried it and indeed it crashes, for this PR is fine but there should be something to prevent those...

Reply:
I think it depends on how you want to use the collection. Indexing into it will crash, because it is effectively empty. Adding new things to it might work, but I haven't tried it.

Reply:
[updated docstring](https://github.com/AIDASoft/podio/pull/447/files#diff-eaff5706c27751eee9136bf2594ec21b3112a331d85ea76b5c7afc69de298a9dR134-R137)

Reply:
I feel like here going out of range may be more of a problem than in C++ (this is not related to this PR but just indexing out of range a `std::deque`):
``` python
event = Frame()
hits = ExampleHitCollection()
hits[0] # seg fault
hits[1] # <cppyy.gbl.MutableExampleHit object at 0x55dcd5d30d50>
hits[1].x() # seg fault
hits.create(0, 1, 2. 3, 4)
hits[0] # <cppyy.gbl.MutableExampleHit object at 0x55dcc4324040>
hits[1] # <cppyy.gbl.MutableExampleHit object at 0x55dcd5d572b0>
hits[2] # seg fault
```
 It's inconsistent since the behavior is undefined, could be a bit hard to debug when you go out of range and it may seem to work but then it doesn't later on. I'm not sure if there are options, not checking in C++ I think is a design choice since there is also `.at`, but in python no one is going to use that probably

Reply:
Yeah in the current form there are not that many handrails in python. Could probably be fixed by another thin python shim that introduces the range check. That could then also be used to raise an error in the *use-after-move* case.

Created #447 to keep track of this.